backend/inspirehep/records/marshmallow/literature/latex.py

```diff
@@ -41,7 +42,8 @@ def get_author_names(self, data):
         return [name.replace(". ", ".~") for name in author_names]
 
     def get_publication_info(self, data):
-        publication_info = BibTexCommonSchema.get_best_publication_info(data)
+        processed_data = deepcopy(data)
+        publication_info = BibTexCommonSchema.get_best_publication_info(processed_data)
```

Comment:
so `BibTexCommonSchema.get_best_publication_info` mutates its argument? That's very misleading. I think you should fix that if it's the case rather than do a `deepcopy` here.

Reply:
no, ```get_best_publication_info``` is not mutating its data. The data is mutated afterwards adding ```doc_type``` and ```texkey```. I should actually maybe copy the publication_info not the entire data